src/transport_udp_protocol.h

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+namespace qtransport {
+    namespace UdpProtocol {
+        /* ------------------------------------------------------------------------
+         * Wire messages
+         * ------------------------------------------------------------------------
+         */
+        constexpr uint8_t PROTOCOL_VERSION = 1;
+
+        /**
+         * @brief UDP Protocol Types
+         * @details Each UDP packet is encoded with a common header, which includes a type.
+         */
+        enum class ProtocolType : uint8_t {
+            CONNECT = 0,
+            DISCONNECT = 1,
+            KEEPALIVE = 2,
+            DATA = 3,
+            DATA_DISCARD = 4,
+            REPORT = 5,
+        };
+
+        /**
+         * @brief UDP Protocol Common Header
+         * @brief Every UDP packet starts with this common header. The data that follows is defined by the type
+         */
+        struct CommonHeader {
+            uint8_t version {PROTOCOL_VERSION};       /// Protocol version
+            ProtocolType type;                        /// Indicates this is a peering message
+        };
+
+        /**
+         * @brief Connect Message
+         *
+         * @details UDP Protocol starts off with a connect message. Messages will be discarded by the remote
+         *      until the new connection sends a connect message.
+         */
+        struct ConnectMsg : CommonHeader {
+            ConnectMsg() { type = ProtocolType::CONNECT; }
+
+            uint16_t idle_timeout { 120 };            /// Idle timeout in seconds. Must not be zero
+
+        } __attribute__((__packed__, aligned(1)));
```

Comment:
Ah, this explains what I was asking about before. This does ensure the size is precisely the size of the object. Why are these not serialized using the message buffer stuff used elsewhere in libquicr? Perhaps because this is one level down from that, but you know this will not be portable.  It would be trivial to create a miniature message serializer for these, since they're all trivial.

Reply:
We can update it if it's really a problem, but:

1. It's portable with clang and gcc of versions that support c++20, which we require.
2. `#pragma pack(1)` is portable with clang, gcc, and vs, so we could use that as well, but that's header wide